src/test/java/org/opensearch/security/SnapshotRestoreTests.java

```diff
@@ -231,106 +235,145 @@ public void testSnapshotCheckWritePrivileges() throws Exception {
+
+        final List<String> listOfIndexesToTest = Arrays.asList("foo", "bar", "baz");
+
+        final Settings settings = Settings.builder()
+                .putList("path.repo", repositoryPath.getRoot().getAbsolutePath())
+                .build();
+
+        setup(Settings.EMPTY, new DynamicSecurityConfig().setSecurityActionGroups("action_groups_packaged.yml"), settings, true, currentClusterConfig);
+
+        try (Client tc = getClient()) {
+            for (String index : listOfIndexesToTest) {
+                tc.admin().indices().create(new CreateIndexRequest(index)).actionGet();
+                tc.index(new IndexRequest(index).setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE).id("document1").source("{ \"foo\": \"bar\" }", XContentType.JSON)).actionGet();
+            }
+        }
+
+       try (Client tc = getClient()) {
+            tc.admin().cluster().putRepository(new PutRepositoryRequest("all").type("fs").settings(Settings.builder().put("location", repositoryPath.getRoot().getAbsolutePath() + "/all"))).actionGet();
+            tc.admin().cluster().createSnapshot(new CreateSnapshotRequest("all", "all_1").indices(listOfIndexesToTest).includeGlobalState(false).waitForCompletion(true)).actionGet();
+       }
+
+        RestHelper rh = nonSslRestHelper();
+
+        Assert.assertEquals(HttpStatus.SC_OK, rh.executePostRequest("_snapshot/all/all_1/_restore?wait_for_completion=true","{\"indices\": \"b*,-bar\", \"rename_pattern\": \"(.+)\", \"rename_replacement\": \"wild_first_restored_index_$1\"}", encodeBasicHeader("nagilum", "nagilum")).getStatusCode());
+        Assert.assertEquals(HttpStatus.SC_OK, rh.executePostRequest("_snapshot/all/all_1/_restore?wait_for_completion=true","{\"indices\": \"-bar,b*\", \"rename_pattern\": \"(.+)\", \"rename_replacement\": \"neg_first_restored_index_$1\"}", encodeBasicHeader("nagilum", "nagilum")).getStatusCode());
+        String wild_first_body = rh.executePostRequest("_snapshot/all/all_1/_restore?wait_for_completion=true","{\"indices\": \"b*,-bar\", \"rename_pattern\": \"(.+)\", \"rename_replacement\": \"wild_first_restored_index_$1\"}", encodeBasicHeader("nagilum", "nagilum")).getBody();
+        assert(!wild_first_body.contains("wild_first_restored_index_foo"));
+        assert(!wild_first_body.contains("wild_first_restored_index_bar"));
+        assert(wild_first_body.contains("wild_first_restored_index_baz"));
+        String neg_first_body = rh.executePostRequest("_snapshot/all/all_1/_restore?wait_for_completion=true","{\"indices\": \"-bar,b*\", \"rename_pattern\": \"(.+)\", \"rename_replacement\": \"negate_first_restored_index_$1\"}", encodeBasicHeader("nagilum", "nagilum")).getBody();
+        assert(!neg_first_body.contains("negate_first_restored_index_foo"));
+        assert(!neg_first_body.contains("negate_first_restored_index_bar"));
+        assert(neg_first_body.contains("negate_first_restored_index_baz"));
```

Comment:
Nit: `assertThat` provides much cleaner output, and IMO is also much easier to read

```java
assertThat(neg_first_body, not(containsString("negate_first_restored_index_foo")));
assertThat(neg_first_body, containsString("negate_first_restored_index_baz"));
```

More reasons I like these matchers https://stackoverflow.com/a/1701143/533057